### PR TITLE
Add some ToJSON instances needed by cardano-node

### LINKED ISSUE #3418: Move ToJSON ledger instances from `cardano-node`
We would like to replace these functions in `cardano-node` with `ToJSON` instances:

https://github.com/input-output-hk/cardano-node/blob/master/cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs#L311-L334

### PR DESCRIPTION
**Pull request by neilmayhew**
# Description

These are currently orphan instances in `cardano-node` but belong in ledger.

Closes #3418

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

**Comment by neilmayhew** [comment]
@Jimbo4350 suggested I add some golden tests to make sure nothing breaks, so I'll do that next

**Comment by neilmayhew** [comment]
The golden tests I've added are the same as the ones I added to node in IntersectMBO/cardano-node#5772. I'm assuming that the ones there will be discarded after being used to check the upgrade after integration. They pass with the instances that are currently in node, so they will verify that nothing has changed.

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs [hunk 1]
```diff
@@ -347,6 +350,15 @@ instance DecCBOR FailureDescription where
     1 -> SumD PlutusFailure <! From <! From
     n -> Invalid n
 
+instance ToJSON FailureDescription where
+  toJSON (PlutusFailure t _bs) =
+    kindObject
+      "FailureDescription"
+      [ "error" .= Aeson.String "PlutusFailure"
+      , "description" .= t
+      -- , "reconstructionDetail" .= bs
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs]: We can add a comment on why this is commented out:
```suggestion
      -- Plutus context cane be pretty big, therefore it is omitted in JSON
      -- , "reconstructionDetail" .= bs
```

**Review by lehins — APPROVED** [review]
Looks good. Thank you.

File: eras/alonzo/test-suite/golden/FailureDescription.json
Comment (lehins) [inline eras/alonzo/test-suite/golden/FailureDescription.json:1]: Almost forgot.

Make sure to add these golden files to the cabal file:

https://github.com/IntersectMBO/cardano-ledger/blob/20b34e4a4b6ee36557bd06d9d10d509d0ea9ac88/eras/alonzo/test-suite/cardano-ledger-alonzo-test.cabal#L17

Comment (neilmayhew) [reply eras/alonzo/test-suite/golden/FailureDescription.json:1]: Good catch! Thanks

**Pull request merged**